From 29eb56d4d16c9a5544ba2b1da2369a9ed3190bb7 Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Sun, 5 Feb 2023 16:49:04 +0100
Subject: [PATCH] [dashboard] Skip task switcher

CCBUG: 465303
(cherry picked from commit 813dcc46c125fbdff4a208a23cec0ae1363f7cd8)
---
 applets/kicker/plugin/dashboardwindow.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/applets/kicker/plugin/dashboardwindow.cpp b/applets/kicker/plugin/dashboardwindow.cpp
index fc6baedcb8..ac7522aab8 100644
--- a/applets/kicker/plugin/dashboardwindow.cpp
+++ b/applets/kicker/plugin/dashboardwindow.cpp
@@ -124,12 +124,12 @@ bool DashboardWindow::event(QEvent *event)
     if (event->type() == QEvent::Expose) {
         // FIXME TODO: We can remove this once we depend on Qt 5.6.1+.
         // See: https://bugreports.qt.io/browse/QTBUG-26978
-        KWindowSystem::setState(winId(), NET::SkipTaskbar | NET::SkipPager);
+        KWindowSystem::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
     } else if (event->type() == QEvent::PlatformSurface) {
         const QPlatformSurfaceEvent *pSEvent = static_cast<QPlatformSurfaceEvent *>(event);
 
         if (pSEvent->surfaceEventType() == QPlatformSurfaceEvent::SurfaceCreated) {
-            KWindowSystem::setState(winId(), NET::SkipTaskbar | NET::SkipPager);
+            KWindowSystem::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
         }
     } else if (event->type() == QEvent::Show) {
         updateTheme();
-- 
GitLab

